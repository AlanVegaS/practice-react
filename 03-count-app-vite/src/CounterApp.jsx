import { useState } from 'react';
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => { 
    const [ counter, setCounter ] = useState(value)

    const addCounter = () => {
        setCounter( c => c + 1)
    }
    const subCounter = () => {
        setCounter( counter - 1 )
    }
    const reset = () => {
        setCounter(value)
    }

    return <><h1>Counter App</h1>
    <h2>{ counter }</h2>
    <button onClick={ addCounter }>+1</button>
    <button onClick={ subCounter }>-1</button>
    <button onClick={ reset }>Reset</button></>
}

CounterApp.prototype = {
    value: PropTypes.number.isRequired
}