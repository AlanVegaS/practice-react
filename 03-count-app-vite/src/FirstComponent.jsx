import PropTypes from "prop-types"

const newMessage = () => {
    return 'Hola, soy Alan'
}

export const ComponentName = ({ tittle, subtittle }) => {
    return <><h1 data-testid = "test-title">{tittle}</h1>
        <h2>{subtittle}</h2>
        <h2>{subtittle}</h2>
        <p>{newMessage()}</p></>
}

ComponentName.propTypes = {
    tittle: PropTypes.string.isRequired,
    subtittle: PropTypes.string,
}
ComponentName.defaultProps = {
    subtittle: 'Hola, soy el subtitulo',
}