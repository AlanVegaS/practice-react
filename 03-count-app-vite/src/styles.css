html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    width: 6rem;
    padding: .5rem;
    font-size: 1.5rem;
    font-weight: 600;
    border-radius: 10px;
    border: none;
    margin: 5px;
    color: white;
    background: linear-gradient(hsl(162, 65%, 46%), hsl(161, 79%, 32%));
}