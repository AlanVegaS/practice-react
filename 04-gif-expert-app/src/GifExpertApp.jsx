import { useState } from "react"
import { AddCategory } from "./components/addCategory"
import { GifGrid } from "./components/GifGrid"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Shark', 'Octopus'])
    const onAddCategory = (newCategory) => {
        console.log(newCategory);
        if (categories.includes(newCategory)) return
        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory //setCategories={setCategories} categories={categories} 
                onNewCategory={onAddCategory}
            />
            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }
        </>
    )
}