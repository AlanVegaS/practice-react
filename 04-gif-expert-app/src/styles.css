body {
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    background-color: rgb(245, 250, 253);
    padding: 60px;
}

h1 {
    text-align: center;
}

input {
    padding: .4rem;
    display: block;
    margin:  3rem auto 3rem ;
    width: 50rem;
    font-weight: 500;
    font-size: 1.1rem;
    border-radius: 5px;
    border: none;
}

.img-section {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    grid-template-rows: 1fr;
    gap: 30px;
}

.card {
    background-color: rgb(226, 226, 226);
    border-radius: 8px;
    overflow: hidden;
    transition: all .2s;
}
.card:hover {
    box-shadow: 2px 2px 10px rgba(163, 163, 163, 0.2);
    transform: scale(1.02);
    cursor: pointer;
}
.card img {
    width: 100%;
}
.card p {
    padding-left: 1rem;
}