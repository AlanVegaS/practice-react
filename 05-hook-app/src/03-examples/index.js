export * from "./BlockQuotes"
export * from "./Loading"
export * from "./multipleCustomHooks"