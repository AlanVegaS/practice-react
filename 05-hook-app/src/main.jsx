//import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
//import { MultipleCustomHooks } from './03-examples/multipleCustomHooks'
//import { FocusScreen } from './04-useRef/FocusScreen'
//import { Layout } from './05-useLayoutEfect/Layout'
//import { Memorize } from './06-memos/Memorize'
//import { MemoHook } from './06-memos/MemoHook'
//import { CallBackHook } from './06-memos/CallBackHook'
//import { Padre } from './07-tarea-memo/Padre'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { HooksApp } from './HooksApp'
//import { HooksApp } from './HooksApp'
//import { CounterApp } from './01-useState/CounterApp'
//import './08-useReducer/intro-reducer'
import { TodoApp } from './08-useReducer/TodoApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <TodoApp/>
  //</React.StrictMode>,
)