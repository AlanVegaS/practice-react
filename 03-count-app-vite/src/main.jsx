//import { createRoot } from 'react-dom/client';
//import { saludo } from './HelloWorld';
//import  {ComponentName}  from './FirstComponent';
import './styles.css';
import ReactDOM from 'react-dom';
import { CounterApp } from './CounterApp';

/*ReactDOM.render(
    <ComponentName tittle="Hola, soy el tÃ­tulo" subtittle = "Soy un sub"/>,
    document.getElementById('root')
);*/
ReactDOM.render(
    <CounterApp value={20}/>,
    document.getElementById('root')
);