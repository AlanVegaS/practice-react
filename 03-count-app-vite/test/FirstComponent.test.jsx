import { getAllByText, getByText, render } from "@testing-library/react"
import { ComponentName } from "../src/FirstComponent"

describe('Pruebas en FirstComponent', () => { 
    /*test('Debe de hacer match con el snapshot', () => { 
        const title = 'Hola soy titulo'
        const {container} = render(<ComponentName tittle = {title} subtittle = "Soy sub"/>)

        expect(container).toMatchSnapshot()
     })*/
     test('Debe mostrar el titulo en un H1', () => { 
        const title = 'Hola soy titulo'
        const {container, getByText, getByTestId} = render(<ComponentName tittle = {title} subtittle = "Soy sub"/>)
        expect(getByText(title)).toBeTruthy()
        
        const h1 = container.querySelector('h1')
        //expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);
        expect(getByTestId('test-title').innerHTML).toBe(title)
    })
    test('Debe mostrar el subtitulo enviado por props', () => { 
        const title = 'Hola soy titulo'
        const subtitle = 'Sub'
        const {getAllByText} = render(<ComponentName tittle = {title} subtittle = {subtitle}/>)

        expect(getAllByText(subtitle).length).toBe(2) 
    })
 })