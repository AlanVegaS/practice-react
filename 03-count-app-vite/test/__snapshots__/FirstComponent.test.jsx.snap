// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstComponent Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy titulo
  </h1>
  <h2>
    Soy sub
  </h2>
  <p>
    Hola, soy Alan
  </p>
</div>
`;
